### PR TITLE
closes #132,#133

### LINKED ISSUE #132: Code generation for Interfaces not working correctly
Hi,

first of all - thank you very much for this great piece of work.
I have some problems with code generation. Attached you find my schema.json and my operations.gql so you can reproduce this.

First, types that implement interfaces are not correctly used in the generated operations.py:
image

Steps to reproduce:
1. `sgqlc-codegen schema schema.json schema.py`
2. `sgqlc-codegen operation --schema schema.json client:schema operations.py operations.gql`
3. Open operations.py and note that `BarBaz` cannot be found. Either an import is missing or the codegen should put out `client.BarBaz` instead.

Second, is it possible to get auto-completion in vscode for the generated operations? Currently the derived type seems to be `Any`. I'm currently using pylance but can of course use something else.

image

[client.zip](https://github.com/profusion/sgqlc/files/6166602/client.zip)

### PR DESCRIPTION
**Pull request by maaft**

### CODE REVIEW

File: sgqlc/codegen/operation.py [hunk 1]
```diff
@@ -456,9 +458,9 @@ def validate_required_arguments(self, node):
             if a['type']['kind'] == 'NON_NULL' and a['defaultValue'] is None:
                 required.add(a['name'])
 
-        for _, var in node.arguments:
-            if var.name in required:
-                required.remove(var.name)
+        for name, _ in node.arguments:
```
Comment (barbieri) [inline sgqlc/codegen/operation.py:461]: could you check if this works with variables? See this fix I had to do sometime ago in a similar code e4a22f4f9b7d4bdf953673d1263930106f4195f1

I'd suggest to keep the old behavior if it's a variable, then:  `name, maybe_var` and you check `hasattr(maybe_var, 'name')`  to use it overriding `name`

File: sgqlc/codegen/operation.py [hunk 2]
```diff
@@ -262,7 +262,9 @@ def format_selection_set_inline_fragment(self, parent, type_condition,
                                              children, lines, idx):
         sel = self.selection_name(parent, '_as__%s' % type_condition, idx)
         idx += 1
-        lines.append('%s = %s.__as__(%s)' % (sel, parent, type_condition))
+        full_type_condition = str(self.schema_name) + "." + type_condition
```
Comment (barbieri) [inline sgqlc/codegen/operation.py:265]: We should use `format_type_usage()` instead, I don't think we should need the `wrapper` version, but that would be a declaration error, on our side (code generation), we better use it to keep consistent.

**Comment by barbieri** [comment]
Hi @maaft , thanks for your patches.

I'm bit picky with the commits and git history, so I'd kindly ask for you to create 2 commits, one with the fix to #132 and another for #133. They should be in the same PR, but only 2 commits, each already incorporating the style fixes.

For #132, I have a suggestion to use `format_type_usage()`, but it's minor.

However for #133, I think we need to cope with both cases... I thought I fixed it... and I did, but I forgot to merge, see https://github.com/profusion/sgqlc/pull/126 :-( my bad

**Comment by barbieri** [comment]
@maaft sorry about the inconvenience, #126 is now merged. You just have to address the other bit #132

**Comment by barbieri** [comment]
also, please git-rebase on the newest master

File: sgqlc/codegen/operation.py (hunk 2 above)
Comment (maaft) [reply sgqlc/codegen/operation.py:265]: Just so I understand you correctly: You suggest to use another `format_type_usage()` instead, e.g. `format_type_condition()`?

Comment (barbieri) [reply sgqlc/codegen/operation.py:265]: There is a method called `self.format_type_usage(typ)` that will handle the schema name prefix.

**Comment by barbieri** [comment]
closing this one as it was fixed in b208ca63b3339f2e6d1be74e8d6efcf304392a0c

please reopen if something is still missing

**Pull request closed without merging**